core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -288,7 +290,8 @@ private[log] class Cleaner(val id: Int,
                            dupBufferLoadFactor: Double,
                            throttler: Throttler,
                            time: Time,
-                           checkDone: (TopicAndPartition) => Unit) extends Logging {
+                           checkDone: (TopicAndPartition) => Unit,
+                           brokerCompressionType: String) extends Logging {
```

Comment:
We can use "log.config.compressionType"  to access particular topic's compression type.